unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Common/AssetPromise.cs

```diff
@@ -127,7 +133,11 @@ protected void OnReuseFinished()
 
         protected void OnLoadSuccess()
         {
-            if (AddToLibrary())
+            addToLibraryCoroutine = CoroutineUtils.StartThrottledCoroutine(AddToLibrary(OnAddToLibrary), OnLoadFailure, throttlingCounter.EvaluateTimeBudget);
```

Comment:
We should stop any current coroutine on-going. Not sure how likely is that edge case, but it could lead to really nasty issues.